src/dc_receive_imf.rs

```diff
@@ -2569,7 +2569,11 @@ mod tests {
         let msg = Message::load_from_db(&t.ctx, msg_id).await.unwrap();
 
         assert_eq!(msg.state, MessageState::OutFailed);
-        assert_eq!(msg.error, error_msg);
+
+        assert_eq!(
+            msg.error(),
+            Some(error_msg.to_string()).filter(|error| !error.is_empty())
```

Comment:
Not sure what's the point of `filter` here, looks like it was copied from `message.rs`.

Reply:
If the error is empty it will use None. I will change the test cases instead to directly set None/Some(s)